cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -912,19 +912,191 @@ def create_image(self, name):
         raise NotImplementedError(
             'To be implemented after GCEVolumeService.')
 
+    @property
+    def _existing_target_instance(self):
+        self_url = self._provider.parse_url(self._gce_instance['selfLink'])
+        try:
+            response = (self._provider.gce_compute
+                                      .targetInstances()
+                                      .list(project=self._provider.project_name,
+                                            zone=self_url.parameters['zone'])
+                                      .execute())
+            if 'items' not in response:
+                return None
+            for target_instance in response['items']:
+                url = self._provider.parse_url(target_instance['instance'])
+                if url.parameters['instance'] == self.name:
+                    return target_instance
+        except Exception as e:
+            cb.log.warning('Exception while listing target instances: %s', e)
+        return None
+
+    @property
+    def _target_instance(self):
+        existing_target_instance = self._existing_target_instance
```

Comment:
It is not needed, but it caches the result of `self._existing_target_instance`, which may save an API call (note that `self._existing_target_instance` may try to list target instances).